### PR TITLE
DOC Document new CMSEditLink extension.

### PR DESCRIPTION
**Pull request by GuySartorelli**
Documents the new extension in https://github.com/silverstripe/silverstripe-admin/pull/1361
Assumes https://github.com/silverstripe/silverstripe-cms/pull/2778 is also merged.

## Related Issue
- https://github.com/silverstripe/silverstripe-admin/issues/1354

### CODE REVIEW

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+# Managing Records
+
+Most records in Silverstripe CMS are managed [in a GridField](../forms/field_types/gridfield) - whether in the [GridField](api:SilverStripe\Forms\GridField\GridField)
+of some other record or directly [in a ModelAdmin](../customising_the_admin_interface/modeladmin/). The notable exceptions to this are
+[SiteConfig](api:SilverStripe\SiteConfig\SiteConfig) and [SiteTree](api:SilverStripe\CMS\Model\SiteTree).
+
+## Getting an edit link
+
+To get a link for editing a record, other than some exceptions like `SiteTree`, you have historically had to generate the link mostly
```
Comment (emteknetnz) [inline en/02_Developer_Guides/00_Model/13_Managing_Records.md:15]: I don't think we need to provide a historical background. The docs we provide are always for the latest version. Unfortunately we don't contain docs for older minor versions, though that's just a limitation of our documentation system, and not something we should attempt to handle here

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+from scratch with mostly hardcoded paths. This was especially tedious with nested `GridField` setups.
+
+As of Silverstripe CMS 4.12.0 there is a new [CMSEditLinkExtension](api:SilverStripe\Admin\CMSEditLinkExtension) specifically
+for the purpose of generating links to the edit forms of records. It operates on the assumption that your record is being edited in
+a [GridFieldDetailForm](../forms/field_types/gridfield#gridfielddetailform) in some `GridField` (be it on another record or in a
+`ModelAdmin`).
+
+When using this extension, your model must also declare its `cms_edit_owner` as a
+[configuration property](../configuration/configuration/#configuration-properties). The value must either be the class name of the
+`ModelAdmin` that directly manages the record, or the `has_one` relation for the record that this model is edited on.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/00_Model/13_Managing_Records.md]:
```suggestion
`ModelAdmin` that directly manages the record, or the `has_one` relation for the record that this model is edited on, which is often the parent DataObject
```

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md (hunk 1 above)
Comment (emteknetnz) [inline en/02_Developer_Guides/00_Model/13_Managing_Records.md]: I don't think we need to provide a historical background. The docs we provide are always for the latest version. Unfortunately we don't contain docs for older minor versions, though that's just a limitation of our documentation system, and not something we should attempt to handle here

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+If the `cms_edit_owner` is a `has_one` relation, the class on the other end of the relation _must_ have
+a reciprocal `has_many` relation as documented in [Relations](./relations#has-many). For best results, use dot notation on the
+`has_many` relation. It must also implement a [getCMSEditLinkForManagedDataObject()](api:SilverStripe\Admin\CMSEditLinkExtension::getCMSEditLinkForManagedDataObject())
+method. The easiest way to do that is for it to have `CMSEditLinkExtension` applied as well.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/00_Model/13_Managing_Records.md]:
```suggestion
method. The easiest way to do that is for it to apply the `CMSEditLinkExtension` to the reciprocal class.
```

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    private static $extensions = [
+        CMSEditLinkExtension::class,
+    ];
+}
+```
+
+[hint]
+If the `cms_edit_owner` is in some vendor dependency that you don't control, you can always apply `CMSEditLinkExtension`
+and the `cms_edit_owner` via yaml.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/00_Model/13_Managing_Records.md]:
```suggestion
and the `cms_edit_owner` via yml.
```

File: en/04_Changelogs/4.12.0.md [hunk 5]
```diff
@@ -111,11 +111,18 @@ SilverStripe\Assets\File:
...
 - [getLinkForModelClass()](api:SilverStripe\Admin\ModelAdmin::getLinkForModelClass()) returns a link to the tab for a specific class in the `ModelAdmin` - useful as a base for links to specific actions.
 - [getLinkForModelTab()](api:SilverStripe\Admin\ModelAdmin::getLinkForModelTab()) similar to `getLinkForModelClass()`, except you pass in a specific model tab you want a link to. This is useful if your `ModelAdmin` has multiple tabs for managing a single class.
 - [isManagedModel](api:SilverStripe\Admin\ModelAdmin::isManagedModel()) returns `true` if the class or model tab passed in is managed by this `ModelAdmin`.
 EditForm
+
+### New `CMSEditLinkExtension`
+
+The silverstripe/admin module has a new [CMSEditLinkExtension](api:SilverStripe\Admin\CMSEditLinkExtension) which makes it trivial to generate an edit link for any `DataObject` even in nested `GridField` setups. This is useful, for example, when [making a previewable `DataObject`](/developer_guides/customising_the_admin_interface/preview/) or as links in reminder emails for stale content.
```
Comment (emteknetnz) [inline en/04_Changelogs/4.12.0.md]:
```suggestion
The `silverstripe/admin module` has a new [CMSEditLinkExtension](api:SilverStripe\Admin\CMSEditLinkExtension) which is used to generate CMS edit links for `DataObject`s, including those in nested `GridField` setups.

This allows you to share DataObject edit links directly with others. This extension is also used to facilitate CMS preview of DataObjects
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/00_Model/13_Managing_Records.md]: Removed.

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/00_Model/13_Managing_Records.md]: Done

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md (hunk 3 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/00_Model/13_Managing_Records.md]: Done

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md (hunk 4 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/00_Model/13_Managing_Records.md]: Done

File: en/04_Changelogs/4.12.0.md (hunk 5 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/4.12.0.md]: Done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**